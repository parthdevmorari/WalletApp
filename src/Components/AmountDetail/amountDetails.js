
import { View, Text, Image } from 'react-native'
import React from 'react'
import styles from './amountDetailsStyle'

const AmountDetails = (props) => {
	const { title,balanceTitle,amount, imageSource,amountSymbol} = props;

	return (
		<View style={styles.container}>
			<View style={[styles.logoContainer]}>
				<Image source={imageSource} style={[styles.logoStyle]} />
			</View>
			<View style={[styles.innerContainer]}>
			<Text style={[styles.txt]}>{title}</Text>
			<Text style={[styles.txt2]}>{balanceTitle}</Text>
			<View style={[styles.amountContainer]}>
				<View style={[styles.box]}>
					<Text style= {styles.symbolStyle}>{amountSymbol}</Text>
					</View>
				<Text style={[styles.Amounttxt]}>{amount}</Text>
			</View>
			</View>
		</View>
	)
}

export default AmountDetails