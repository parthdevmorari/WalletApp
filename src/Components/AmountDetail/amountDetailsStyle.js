
import { StyleSheet, StatusBar } from 'react-native';
import { widthPercentageToDP as wp, heightPercentageToDP as hp } from 'react-native-responsive-screen';
import { color } from '../../Constants/Color'
const styles = StyleSheet.create({
	container:{
		paddingHorizontal:wp(6),
		paddingTop:wp(2.5)
	},

	logoContainer : {
		height: wp(7.5),
		width: wp(7.5),
		marginRight:wp(2.5),
		marginTop: wp(5),
		alignSelf:'flex-end'
	},
	logoStyle: {
		width: "100%",
		height: "100%",
	},
	txt: {
		fontWeight: "bold",
		fontSize: wp(6),
		color: color._WHITE,
		marginBottom: wp(10)
	},
	txt2: {
		fontSize: wp(3.5),
		color: color._WHITE,
		textAlign: "left",
		fontWeight: "500",
		
	},
	amountContainer:{
		flexDirection: "row",
		justifyContent: 'flex-start',
		alignItems: "center",
		marginTop:wp(2.5)
		

	},
	box: {
		width: wp(10),
		height: wp(6),
		backgroundColor:color._01D167,
		borderRadius: wp(1), 
	},
	symbolStyle: {
		fontSize: wp(3.5),
		fontWeight: "bold",
		color: color._WHITE,
		textAlign: "center",
		marginTop: wp(0.75)
	},
	Amounttxt:{
		fontSize: wp(6),
		fontWeight: "bold",
		color: color._WHITE,
		marginLeft: wp(2.75),
		textAlignVertical:'center',

	},


});
export default styles;