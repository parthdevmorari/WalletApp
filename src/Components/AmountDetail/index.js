import component from './amountDetails';
export default component;