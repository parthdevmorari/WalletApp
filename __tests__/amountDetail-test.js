import 'react-native';
import React from 'react';
import AmountDetail from '../src/Components/AmountDetail'
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});

describe('Amount Detail Component', () => {
  test('Amount Detail Component', () => {
    const wrapper = shallow(<AmountDetail />);
    expect(wrapper.exists()).toBe(true);
  });
  it('Amount Detail  toMatchSnapshot', () => {
    const wrapper = shallow(<AmountDetail />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});