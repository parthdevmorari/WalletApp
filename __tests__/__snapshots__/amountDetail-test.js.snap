// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Amount Detail Component Amount Detail  toMatchSnapshot 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 45,
      "paddingTop": 19,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-end",
          "height": 56.5,
          "marginRight": 19,
          "marginTop": 37.5,
          "width": 56.5,
        },
      ]
    }
  >
    <Image
      style={
        Array [
          Object {
            "height": "100%",
            "width": "100%",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 45,
            "fontWeight": "bold",
            "marginBottom": 75,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 26.5,
            "fontWeight": "500",
            "textAlign": "left",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginTop": 19,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#01D167",
              "borderRadius": 7.5,
              "height": 45,
              "width": 75,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 26.5,
              "fontWeight": "bold",
              "marginTop": 5.5,
              "textAlign": "center",
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 45,
              "fontWeight": "bold",
              "marginLeft": 20.5,
              "textAlignVertical": "center",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`Button Component Button  toMatchSnapshot 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 45,
      "paddingTop": 19,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-end",
          "height": 56.5,
          "marginRight": 19,
          "marginTop": 37.5,
          "width": 56.5,
        },
      ]
    }
  >
    <Image
      style={
        Array [
          Object {
            "height": "100%",
            "width": "100%",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 45,
            "fontWeight": "bold",
            "marginBottom": 75,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 26.5,
            "fontWeight": "500",
            "textAlign": "left",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginTop": 19,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#01D167",
              "borderRadius": 7.5,
              "height": 45,
              "width": 75,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 26.5,
              "fontWeight": "bold",
              "marginTop": 5.5,
              "textAlign": "center",
            }
          }
        />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 45,
              "fontWeight": "bold",
              "marginLeft": 20.5,
              "textAlignVertical": "center",
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
